src/codec.rs

```diff
@@ -147,6 +147,43 @@ impl Codec for u64 {
     }
 }
 
+impl Codec for String {
+    fn encode(&self, buffer: &mut Vec<u8>) -> Result<(), CodecError> {
+        let string_bytes = self.as_bytes();
+        let string_bytes_len = match u16::try_from(string_bytes.len()) {
+            Ok(v) => v,
+            Err(_) => return Err(CodecError::EncodingError),
+        };
+        string_bytes_len.encode(buffer)?;
+        buffer.extend_from_slice(&string_bytes);
+        Ok(())
+    }
+
+    fn decode(cursor: &mut Cursor) -> Result<Self, CodecError> {
+        let string_length = u16::decode(cursor)?;
+        match std::str::from_utf8(cursor.consume(string_length.into())?) {
+            Ok(s) => Ok(s.to_string()),
+            Err(_) => Err(CodecError::DecodingError),
+        }
+    }
+}
+
+impl<T> Codec for Vec<T>
```

Comment:
So this is essentially an alias for `encode_vec(VecSize::VecU32, ...)`?

I find it somewhat dangerous because it might just be used in places where the vec size is not 32 bit.

Reply:
Yeah the `encode_vec` functions aren't very ergonomic. When redoing encoding we should add the vecsize to the trait functions. More importantly does this allow to encode vectors of vectors without having to handle the length manually.

Reply:
I agree on the ergonomy. I think the ideal solution would be to write a `serde` plug-in, or to use something similar where you can annotate fields of structs about how they should be serialised. And then you can just use `derive` for those structs most of the time. But this is for another day.

Reply:
Right, that's why I put the two implementations in here. Expecting that this will be changed in future anyway.